### ISSUE TITLE
Error on start

### ISSUE THREAD

**Issue by kireerik** [sidorares/hot-module-replacement#5]
I got this error on start 2 times so far:
```bash
hot-module-replacement\index.js:14
       if (module.hot._selfAccepted) {
                      ^

 TypeError: Cannot read property '_selfAccepted' of undefined
     at pathsToAcceptingModules (hot-module-replacement\index.js:14:22)
     at collectDependencies (hot-module-replacement\index.js:33:5)     at Watcher.<anonymous> (hot-module-replacement\index.js:64:20)
     at Watcher.emit (events.js:182:13)
     at callback (node-watch\lib\watch.js:254:14)
     at node-watch\lib\watch.js:112:10
     at Array.forEach (<anonymous>)
     at Timeout.handle [as _onTimeout] (node-watch\lib\watch.js:107:24)
     at listOnTimeout (timers.js:324:15)
     at processTimers (timers.js:268:5)
 exited with code 1
```
I re-started my project without any changes and after that I no longer got this error.

Version: `3.0.1`
Operating system: `Windows`

**Comment by kireerik** [comment]
I got this error a couple of other times on start.

I also found a consistent case:
When you `require` a `json` file from a `module` which can be a top level or a required one. The issue occurs when you modify the `json` file and save the changes.

example:

data.json
```JSON
{}
```

module.js
```JavaScript
require('./data')
````

Modify `data.json` like this for example:
```JSON
{ }
```

**Comment by sidorares** [comment]
This should be fixed in `fix/extensions` branch. Could you test it with your code?

**Comment by sidorares** [comment]
yes, definitely interested! I'll try to merge and publish tomorrow.

**Comment by sidorares** [comment]
published as v3.0.2

**Comment by kireerik** [comment]
Here is the build tool I have mentioned earlier: https://github.com/kireerik/refo

Maybe we can add it to a who is using hot-module-replacement section in the readme if you wish.
